File: session06/GrowArray2.java
// in Java this is called ArrayList
// in C++ this is called vector

public class GrowArray2 {
	private	int[] data;
	private	int used;
	//	int capacity; = data.length
	void grow() {
		// DOUBLE EACH TIME!!!
	}
	private void checkGrow() {
		if (used == capacity)
			grow();
	}
	public void addEnd(int v) { //O(1)
		checkGrow();

	}
	public void addStart(int v); // O(n)

	public 	void removeEnd() { //O(1)

	}
	public String toString() {
		// int = -2100000000
		StringBuilder b = new StringBuilder(11*data.length);
		for (int i = 0; i < data.length; i++)
			b.append(data[i]).append(" ");

	}
	
};

File: GrowArray/GrowArray2.cc
// in Java this is called ArrayList
// in C++ this is called vector

class GrowArray {
private:
	int* data;
	int used;
	int capacity;
	void grow() {
		// DOUBLE EACH TIME!!!
	}
	void checkGrow() {
		if (used == capacity)
			grow();
	}
public:
  void addEnd(int v) { //O(1)
		checkGrow();

	}
	void addStart(int v); // O(n)

	void removeEnd() {

	}
};
